daemon/src/model/cfd.rs

```diff
@@ -700,14 +707,75 @@ pub enum CfdStateChangeEvent {
     CommitTxSent,
 }
 
+/// Returns the Profit/Loss (P/L) as Bitcoin. Losses are capped by the provided margin
 fn calculate_profit(
-    _intial_price: Usd,
-    _current_price: Usd,
-    _interest_per_day: Decimal,
-    _fee: Usd,
-) -> Result<(Amount, Usd)> {
-    // TODO: profit calculation
-    Ok((Amount::ZERO, Usd::ZERO))
+    initial_price: Usd,
+    current_price: Usd,
+    quantity: Usd,
+    margin: Amount,
+    position: Position,
+) -> Result<(Decimal, Decimal)> {
```

Comment:
Returning a tuple with the same type is asking for trouble to mix them up.

The first value is in Bitcoin yes? Can we return that as `bitcoin::Amount` please?

Reply:
> The first value is in Bitcoin yes? Can we return that as bitcoin::Amount please?

`bitcoin::Amount` uses a u64 internally so unfortunately we can't because the value can be negative.

Reply:
Use https://docs.rs/bitcoin/0.27.1/bitcoin/util/amount/struct.SignedAmount.html then.